source/screens/caseScreens/CaseOverview.tsx

```diff
@@ -356,7 +356,7 @@ function CaseOverview(props): JSX.Element {
       // since we cannot react to changes as of now.
       const delayBeforeFetch = 4000;
       wait(delayBeforeFetch).then(fetchCases).catch(onFailedToFetchCases);
-    }, [fetchCases])
```

Comment:
This probably fixes the issue , but overall, the combined logic in CaseOverView.tsx is convoluted and should be refactored.

Reply:
I believe the refactoring is out of the scope of this PR and should be it's own task.